tests/Common/SpanIntegrationChecker.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace DDTrace\Tests\Common;
+
+use DDTrace\Contracts\Span;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Checks that spans are assigned the proper integration.
+ */
+class SpanIntegrationChecker
+{
+    /**
+     * Returns the known matching patterns operation ==> integration class.
+     */
+    public function defineIntegrationsByPattern()
+    {
+        // <regex pattern> => <integration class>
+        return [
+            // Prepend your operation names with custom for custom spans to have the span check disabled
+            '/custom.*/' => null,
+            // Officially supported integrations
+            '/curl_.*/' => 'DDTrace\Integrations\Curl\CurlIntegration',
+            '/Elasticsearch(.).*/' => 'DDTrace\Integrations\ElasticSearch\V1\ElasticSearchIntegration',
+            '/GuzzleHttp.*/' => 'DDTrace\Integrations\Guzzle\GuzzleIntegration',
+            '/laravel.*/' => 'DDTrace\Integrations\Laravel\LaravelIntegration',
+            '/Memcached.*/' => 'DDTrace\Integrations\Memcached\MemcachedIntegration',
+            '/Mongo(Client)|(DB)|(Collection).*/' => 'DDTrace\Integrations\Mongo\MongoIntegration',
+            '/mysqli.*/' => 'DDTrace\Integrations\Mysqli\MysqliIntegration',
+            '/PDO(.)|(Statement).*/' => 'DDTrace\Integrations\PDO\PDOIntegration',
+            '/Predis.*/' => 'DDTrace\Integrations\Predis\PredisIntegration',
+            '/symfony.*/' => 'DDTrace\Integrations\Symfony\SymfonyIntegration',
+            '/web.*/' => 'DDTrace\Integrations\Web\WebIntegration',
+            '/zf.*/' => 'DDTrace\Integrations\ZendFramework\ZendFrameworkIntegration',
+        ];
+    }
+
+    /**
+     * Checks that $span belongs to the proper integration.
+     *
+     * @param TestCase $test
+     * @param Span $span
+     */
+    public function checkIntegration(TestCase $test, Span $span)
```

Comment:
This could be static. :)
```suggestion
    public static function checkIntegration(TestCase $test, Span $span)
```

Reply:
Same as above ^^